lib/code_corps/accounts/user_invites.ex

```diff
@@ -0,0 +1,100 @@
+defmodule CodeCorps.Accounts.UserInvites do
+  alias CodeCorps.{Project, ProjectUser, Repo, User, UserInvite}
+  alias Ecto.{Changeset, Multi}
+
+  import Ecto.Query
+
+  @spec create_invite(map) :: {:ok, UserInvite.t} | {:error, Changeset.t}
+  def create_invite(%{} = params) do
+    %UserInvite{}
+    |> Changeset.cast(params, [:email, :name, :role, :inviter_id, :project_id])
+    |> Changeset.validate_required([:email, :inviter_id])
+    |> Changeset.validate_inclusion(:role, ~w(contributor admin owner))
+    |> Changeset.assoc_constraint(:inviter)
+    |> Changeset.assoc_constraint(:project)
+    |> ensure_email_not_owned_by_member()
+    |> Repo.insert
+  end
+
+  @spec ensure_email_not_owned_by_member(Changeset.t) :: Changeset.t
+  defp ensure_email_not_owned_by_member(%Changeset{} = changeset) do
+    email = changeset |> Changeset.get_change(:email)
+    project_id = changeset |> Changeset.get_change(:project_id)
+
+    case [email, project_id] do
+      [nil, _] -> changeset
+      [_, nil] -> changeset
+      [email, project_id] ->
+        count =
+          ProjectUser
+          |> where(project_id: ^project_id)
+          |> join(:inner, [pu], u in User, pu.user_id == u.id)
+          |> where([_pu, u], u.email == ^email)
+          |> select([pu, _U], count(pu.id))
+          |> Repo.one
+
+        if count > 0 do
+          changeset |> Changeset.add_error(:email, "Already associated with a project member")
+        else
```

Comment:
I think the requirement wasn't quite clear here: we want to be sure that the email is not associated with a `User`.

Reply:
Several questions then

- will a project admin/owner ever be able to invite an existing codecorps user to their project?
  - would that not create a `UserInvite` and create something else? Maybe a `ProjectUser` record with a status of `"invited"`?

- what happens in the following scenario:
  - invite is created for a "mike@mail.com"
  - later, independently, a user registers with that email
  - even later, that same user tries to claim the invite